### PR TITLE
Added language specific page for Golang [resolves #265]

### LINKED ISSUE #265: Programming-Language Specific Pages: Golang
### Prize Bounty
0.3 ETH

### Challenge Description
Let's add resource pages that cater to specific user groups to make Ethereum more approachable to a broader range of people.

Build a dedicated page for Go developers that includes:
- a brief description of Ethereum
- beginner articles
- intermediate articles
- advanced use patterns
- Go projects and tools
- Go communities (e.g. project Gitter channels)

The Python page is a great template to work off:
https://ethereum.org/python/

Also, add a link to the page from the [/developers page "Language Specific Resources" section](https://github.com/ethereum/ethereum-org-website/blob/dev/docs/developers/index.md#language-specific-resources).

### Submission Requirements
The PR must include everything outlined in the challenge description above.

### Submission Deadline
The deadline for submissions is November 11, 2019.

### Judging Criteria
The prize will be awarded to the first valid submission.

### Winner Announcement Date
The submission will be reviewed by November 18. Winner announcements and prize payouts will occur before November 31, 2019.

### PR DESCRIPTION
**Pull request by Abhimanyu121**
## Programming-Language Specific Pages: Golang
Added language-specific page for Golangfollowing https://ethereum.org/python/ page as a template

## Related Issue
Resolves #265 

## Checklist:

- [x] I have read the contributing guidelines in the project README.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/golang/index.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## Beginner resources
+- [Choosing an Ethereum Client](https://www.trufflesuite.com/docs/truffle/reference/choosing-an-ethereum-client)
+- [Use Golang to Connect to Ethereum](https://www.youtube.com/watch?v=-7uChuO_VzM)
+- [Deploy Ethereum Smart Contracts Using Golang](https://www.youtube.com/watch?v=pytGqQmDslE)
+- [A Step By Step Guide To Testing and Deploying Ethereum Smart Contracts in Go](https://hackernoon.com/a-step-by-step-guide-to-testing-and-deploying-ethereum-smart-contracts-in-go-9fc34b178d78)
+- [Go Ethereum book](https://goethereumbook.org/)
+
+
+## Intermediate articles
+- [Official geth Documentation](https://nethereum.readthedocs.io/en/latest/ethereum-and-clients/geth/)
```
Comment (samajammin) [inline docs/golang/index.md]: Your link is to the nethereum doucmentation - I believe this is the official Geth documentation: https://geth.ethereum.org/docs/

Comment (Abhimanyu121) [reply docs/golang/index.md]: I will change it.
Thank you

**Comment by Abhimanyu121** [comment]
Thank you, I will rebase it.

**Comment by Abhimanyu121** [comment]
@samajammin Thank you for your response, I have pushed the updated doc, can you please review it.

**Comment by samajammin** [comment]
@all-contributors please add @Abhimanyu121 for content

**Comment by Abhimanyu121** [comment]
> Thanks for the update! I hope you continue to contribute & curate this page

Yes I would love to.
Thank you 😊

**Pull request merged**